src/plugins/formulas/formulas.js

```diff
@@ -292,43 +543,123 @@ export class Formulas extends BasePlugin {
+   * @param {number} amount Amount of rows to be removed.
+   * @returns {*|boolean} If false is returned the action is canceled.
+   */
   onBeforeRemoveRow(row, amount) {
-    if (!this.hyperformula.isItPossibleToRemoveRows(this.hyperformula.getSheetId(this.sheetName), [row, amount])) {
+    if (!this.hyperformula.isItPossibleToRemoveRows(this.sheetId, [row, amount])) {
       return false;
     }
   }
 
+  /**
+   * `beforeRemoveCol` hook callback.
+   *
+   * @private
+   * @param {number} col Visual index of starter column.
+   * @param {number} amount Amount of columns to be removed.
+   * @returns {*|boolean} If false is returned the action is canceled.
+   */
   onBeforeRemoveCol(col, amount) {
-    if (!this.hyperformula.isItPossibleToRemoveRows(this.hyperformula.getSheetId(this.sheetName), [col, amount])) {
+    if (!this.hyperformula.isItPossibleToRemoveRows(this.sheetId, [col, amount])) {
       return false;
     }
   }
 
+  /**
+   * `afterCreateRow` hook callback.
+   *
+   * @private
+   * @param {number} row Represents the visual index of first newly created row in the data source array.
+   * @param {number} amount Number of newly created rows in the data source array.
+   */
   onAfterCreateRow(row, amount) {
-    this.hyperformula.addRows(this.hyperformula.getSheetId(this.sheetName), [row, amount]);
+    this.hyperformula.addRows(this.sheetId, [row, amount]);
   }
 
+  /**
+   * `afterCreateCol` hook callback.
+   *
+   * @private
+   * @param {number} col Represents the visual index of first newly created column in the data source.
+   * @param {number} amount Number of newly created columns in the data source.
+   */
   onAfterCreateCol(col, amount) {
-    this.hyperformula.addColumns(this.hyperformula.getSheetId(this.sheetName), [col, amount]);
+    this.hyperformula.addColumns(this.sheetId, [col, amount]);
   }
 
+  /**
+   * `beforeRemoveRow` hook callback.
+   *
+   * @private
+   * @param {number} row Visual index of starter row.
+   * @param {number} amount Amount of rows to be removed.
+   * @returns {*|boolean} If false is returned the action is canceled.
+   */
   onAfterRemoveRow(row, amount) {
-    this.hyperformula.removeRows(this.hyperformula.getSheetId(this.sheetName), [row, amount]);
+    this.hyperformula.removeRows(this.sheetId, [row, amount]);
   }
 
+  /**
+   * `afterRemoveCol` hook callback.
+   *
+   * @private
+   * @param {number} col Visual index of starter column.
+   * @param {number} amount An amount of removed columns.
+   */
   onAfterRemoveCol(col, amount) {
-    this.hyperformula.removeColumns(this.hyperformula.getSheetId(this.sheetName), [col, amount]);
+    this.hyperformula.removeColumns(this.sheetId, [col, amount]);
+  }
+
+  /**
+   * HyperFormula's `valuesUpdated` event callback.
+   *
+   * @param {Array} changes Array of objects containing information about HF changes.
+   */
+  onHFvaluesUpdated(changes) {
+    let isAffectedByChange = false;
```

Comment:
Should be refactored to `const`, or not use the variable binding at all.

Reply:
It's being overwritten here, though 
```
    changes.some((change) => {
      isAffectedByChange = change?.address?.sheet === this.sheetId;

      return isAffectedByChange;
    });
```

Reply:
I meant [like this](https://github.com/handsontable/handsontable/pull/7823/files#r619060595)